src/main/host/thread.rs

```
@@ -67,6 +69,91 @@ impl Thread {
+                ThreadShmem::clone(&self.shim_shared_memory, &host_shmem_prot.root);
+
+            // thread id is updated to make this the new thread group leader.
+            thread_shmem.tid = new_tid.into();
+
+            // sigaltstack is reset to disabled.
+            unsafe {
+                *thread_shmem
+                    .protected
+                    .borrow_mut(&host_shmem_prot.root)
+                    .sigaltstack_mut() = stack_t::new(
+                    std::ptr::null_mut(),
+                    linux_api::signal::SigAltStackFlags::SS_DISABLE,
+                    0,
+                )
+            };
+
+            self.shim_shared_memory = shmalloc(thread_shmem);
+        }
+
+        unsafe { c::syscallhandler_free(self.syscallhandler.ptr()) };
+        self.syscallhandler = unsafe {
+            SendPointer::new(c::syscallhandler_new(
+                host.id(),
+                self.process_id.into(),
+                new_tid.into(),
+            ))
+        };
+
+        // Update descriptor table
+        {
+            // Descriptor table is unshared
+            let desc_table_rc = self.desc_table.take().unwrap();
+            let mut desc_table = DescriptorTable::clone(&desc_table_rc.borrow(host.root()));
+            desc_table_rc.explicit_drop(host.root());
+
+            // Any descriptors with CLOEXEC are closed.
+            let to_close: Vec<DescriptorHandle> = desc_table
+                .iter()
+                .filter_map(|(handle, descriptor)| {
+                    if descriptor.flags().contains(DescriptorFlags::FD_CLOEXEC) {
+                        Some(*handle)
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            for handle in to_close {
+                log::trace!("Unregistering FD_CLOEXEC descriptor {handle:?}");
+                drop(desc_table.deregister_descriptor(handle).unwrap());
+            }
```

maybe add a comment that this is ignoring any errors from closing the file object

---

Changed to explicitly close and log errors